src/main/scala/org/constellation/consensus/Round.scala

```diff
@@ -0,0 +1,157 @@
+package org.constellation.consensus
+import akka.actor.{Actor, ActorLogging, Cancellable, Props}
+import cats.implicits._
+import constellation.{wrapFutureWithMetric, _}
+import org.constellation.consensus.EdgeProcessor.{FinishedCheckpoint, FinishedCheckpointResponse}
+import org.constellation.consensus.NodeRemoteSender.{TransactionsProposal, UnionBlockProposal}
+import org.constellation.consensus.Round._
+import org.constellation.primitives.Schema.{CheckpointCacheData, EdgeHashType, TypedEdgeHash}
+import org.constellation.primitives._
+import org.constellation.util.Validation.EnrichedFuture
+import org.constellation.{ConfigUtil, DAO}
+
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.ExecutionContextExecutor
+import scala.concurrent.duration._
+
+class Round(id: RoundId, dao: DAO) extends Actor with ActorLogging {
+
+  val transactionProposals: mutable.Map[FacilitatorId, Seq[Transaction]] = mutable.Map()
+  val checkpointBlockProposals: mutable.Map[FacilitatorId, CheckpointBlock] = mutable.Map()
+
+  var numberOfCheckpointBlockProposalsSent: Int = _
+  var numberOfTransactionProposalsSent: Int = _
+
+  var unionTransactionProposalsTikTok: Cancellable = _
+
+
+  override def preStart(): Unit = {
+    self ! Initialize
+  }
+
+  override def receive: Receive = {
+
+    case Initialize =>
+      dao
+        .pullTransactions(dao.minCheckpointFormationThreshold)
```

Comment:
This needs to be changed to also pull messages, otherwise it will create different blocks during the unioning phase.